src/libasr/asr_utils.h

```diff
@@ -138,7 +138,7 @@ static inline std::string type_to_str(const ASR::ttype_t *t)
             return "logical";
         }
         case ASR::ttypeType::Character: {
-            return "character";
+            return "string";
```

Comment:
In Fortran, there's an intrinsic data type `character` that stores characters and strings, while in Python, `str` is used everywhere.  Since the `libasr` part is shared by LPython and LFortran, I am skeptical about choosing one out of these two.  The error message looks pretty good here, though.
```console
semantic error: Type mismatch in assignment, the types must be compatible
 --> tests/errors/test_assign_type_mismatch.py:5:5
  |
5 |     a = "string"
  |     ^   ^^^^^^^^ type mismatch (integer and string)
```

Reply:
As mentioned in the other issue, we have to actually check more than just the type, we have to check dimension (if it is an array) and length (if it is just one character or a string).

So I would for now keep "character" in libasr, then in LFortran we check the len and return an error with either character or string; in LPython we always say string. Finally, we check dimensions in both, and if it is an array, we say array of strings, or something like that. Or possibly just write the type as one would write it in LPython, so the type is `str`, or possibly `str[5]` for an array (or `list[str]` for a list).

Reply:
We are planning to add Python and Fortran printing from ASR, into libasr. So we'll have to have a function that takes an ASR type, and returns `list[str]` (for LPython) or `character(len=5)` (for LFortran) etc.